File: cassandra-service-base/src/main/scala/com/github/sedovalx/cassandra/services/base/CassandraMapperGeneric.scala
package com.github.sedovalx.cassandra.services.base

import java.util.concurrent.CompletableFuture

import com.datastax.driver.core.Statement
import com.datastax.driver.mapping.Mapper

/**
  * Created by Alexander
  * on 26.12.2015.
  */
trait CassandraMapperGeneric[T] {
    def saveQuery(entity: T): Statement
    def save(entity: T, options: Mapper.Option*): Unit
    def saveAsync(entity: T, options: Mapper.Option*): CompletableFuture[Void]

    def deleteQuery(entity: T): Statement
    def delete(entity: T, options: Mapper.Option*): Unit
    def deleteAsync(entity: T, options: Mapper.Option*): CompletableFuture[Void]
}

File: cassandra-service-base/src/main/java/com/github/sedovalx/cassandra/services/base/CassandraMapperGeneric.java
package com.github.sedovalx.cassandra.services.base;

import com.datastax.driver.core.Statement;
import com.datastax.driver.mapping.Mapper;

import java.util.concurrent.CompletableFuture;

/**
 * Created by Alexander
 * on 22.03.2016.
 */
public interface CassandraMapperGeneric<Entity> {
    Statement saveQuery(Entity entity);
    void save(Entity entity, Mapper.Option... options);
    CompletableFuture<Void> saveAsync(Entity entity, Mapper.Option... options);

    Statement deleteQuery(Entity entity);
    void delete(Entity entity, Mapper.Option... options);
    CompletableFuture<Void> deleteAsync(Entity entity, Mapper.Option... options);
}
